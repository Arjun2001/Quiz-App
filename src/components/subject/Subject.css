.sub-main {
    padding: 10px;
    text-align: center;
    background: white;
    border-radius: 10px;
    width: 100%;
    border: 5px solid #b67162;
}

.cards {
    display: flex;
    flex-direction: row;
    flex-wrap:wrap;
}

.MuiPaper-root {
    border-radius: 15px !important;
    margin: 0px 15px 15px 0px;
}

.click-class {
    cursor: pointer;
}

.makeStyles-paper-13{
    width: 550px;
}

.MuiCardContent-root {
    padding-bottom: 5px !important;
}