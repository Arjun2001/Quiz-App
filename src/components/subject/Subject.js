import React, {useState} from 'react'
import './Subject.css'
import SubCard from './SubCard'
import AddCard from './AddCard'

import { makeStyles } from '@material-ui/core/styles';
import Modal from '@material-ui/core/Modal';
import Backdrop from '@material-ui/core/Backdrop';
import Fade from '@material-ui/core/Fade';
import { Button, Form, FormGroup, Label, Input, FormText } from 'reactstrap';

function Subject() {

    const [open, setOpen] = useState(false);

    const useStyles = makeStyles((theme) => ({
        modal: {
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',
        },
        paper: {
          backgroundColor: theme.palette.background.paper,
          border: '2px solid #000',
          boxShadow: theme.shadows[5],
          padding: theme.spacing(2, 4, 3),
        },
      }));
      const classes = useStyles();
    const handleOpen = () => {
    setOpen(true);
    };

    const handleClose = () => {
    setOpen(false);
    };
    
    return (
        <div className='sub-main'>
            <h3>SUBJECTS</h3>
            <hr></hr>
            <div className="cards">
                <SubCard />
                <AddCard open={handleOpen}/> 
            </div>
            <Modal
                aria-labelledby="transition-modal-title"
                aria-describedby="transition-modal-description"
                className={classes.modal}
                open={open}
                onClose={handleClose}
                closeAfterTransition
                BackdropComponent={Backdrop}
                BackdropProps={{
                timeout: 500,
                }}
            >
                <Fade in={open}>
                <div className={classes.paper}>
                <Form>
                    <FormGroup>
                        <Label for="exampleEmail">Course Code</Label>
                        <Input type="text" name="course_code" id="exampleEmail" />
                    </FormGroup>
                    <FormGroup>
                        <Label for="exampleEmail">Course Name</Label>
                        <Input type="text" name="course_name" id="exampleEmail" />
                    </FormGroup>
                    <FormGroup>
                        <Label for="exampleEmail">Course Photo</Label>
                        <Input type="text" name="course_name" id="exampleEmail"/>
                    </FormGroup>
                    <FormGroup>
                        <Label for="exampleEmail">Course Description</Label>
                        <Input type="text" name="course_name" id="exampleEmail" />
                    </FormGroup>
                    <Button>Submit</Button>
                </Form>
                </div>
                </Fade>
            </Modal>
        </div>
    )
}

export default Subject
