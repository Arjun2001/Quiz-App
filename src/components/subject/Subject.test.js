import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Subject from './Subject'

describe('Blog component', () => {
    it('div rendering', () => {
        const wrapper = shallow(<Subject />);
        const text = wrapper.find('h3').text();
        expect(text).toEqual('SUBJECTS');
    });

    it('div rendering', () => {
        const wrapper = shallow(<Subject />);
        wrapper.find('form');
    });

    it('matches the snapshot', () => {
        const tree = renderer.create(<Subject />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});