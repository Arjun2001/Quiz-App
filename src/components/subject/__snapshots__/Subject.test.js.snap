// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog component matches the snapshot 1`] = `
<div
  className="sub-main"
>
  <h3>
    SUBJECTS
  </h3>
  <hr />
  <div
    className="cards"
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexWrap": "wrap",
        }
      }
    >
      <div />
    </div>
  </div>
</div>
`;
